pkgs/development/libraries/qt-6/patches/qtbase-qmake-qt-prepare-tool-find-qt-tool.sh

```diff
@@ -0,0 +1,70 @@
+# find absolute path of a qt tool
+
+# this script is called from qtbase-dev/mkspecs/features/qt_functions.prf
+
+# resolution order:
+# 1. try $QMAKE_<TOOL_NAME> - example: tool_name=lrelease -> $QMAKE_LRELEASE
+# 2. search in $instloc passed by qt_functions.prf
+# 3. search in all prefixes in $QMAKEPATH
+# 4. search in $PATH
+
+if [[ $# != 2 ]]; then
+    echo 'usage: find-qt-tool.sh tool_name instloc' >&2
+    exit 1
+fi
+
+# example: lrelease
+tool_name=$1
+
+# absolute path to bin/ or libexec/
+instloc=$2
+
+if (( "${NIX_DEBUG:-0}" >= 1 )); then
+  log() {
+    echo "find-qt-tool.sh $tool_name: $*" >&2
+  }
+else
+  log() { :; }
+fi
+
+# test: tool was not found
+#log "not found"; exit
+
+# example: some-tool.pl -> QMAKE_SOME_TOOL_PL
+tool_cmd_env=QMAKE_$(echo $tool_name | tr '[:lower:]-.' '[:upper:]__')
```

Comment:
slashes are not handled, so we cannot override `some/tool.py` with `SOME_TOOL_PY=/other/path/to/tool.py`

probably not needed ...

i didnt check, but all the calls to `qtPrepareTool` should have a simple basename as the first parameter = tool_name

Reply:
slashes are now replaced with _